### PR TITLE
Support new writable option from Corestore

### PR DESCRIPTION
**Pull request by LuKks**
Depends on https://github.com/holepunchto/corestore/pull/54

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -140,10 +137,12 @@ module.exports = class Hyperdrive extends ReadyResource {
...
-      onwait: this._onwait
+      onwait: this._onwait,
+      writable: !this._readonly
     })
     await blobsCore.ready()
 
+    // TODO: readonly related?
     if (this.closing) {
       await blobsCore.close()
       return false
```
Comment (LuKks) [inline index.js:148]: This condition was added in the middle of the `readonly` feature, is this still needed or related? Any hints on how to test it?

File: package.json [hunk 2]
```diff
@@ -35,7 +35,7 @@
     "@hyperswarm/testnet": "^3.0.0",
     "b4a": "^1.6.0",
     "brittle": "^3.1.0",
-    "corestore": "^6.1.1",
+    "corestore": "github:holepunchto/corestore#writable-option",
```
Comment (LuKks) [inline package.json:38]: Needs updating

**Comment by LuKks** [comment]
CI error unrelated (`clearAll` method), will see to fix it

File: index.js (hunk 1 above)
Comment (mafintosh) [reply index.js:148]: thats needed in general yea, otherwise this session would leak

**Pull request merged**